include/boost/math/special_functions/detail/unchecked_factorial.hpp

```diff
@@ -979,6 +979,8 @@ inline T unchecked_factorial(unsigned i)
    return unchecked_factorial_imp<T>(i, tag_type());
 }
 
+#endif // BOOST_NO_LEXICAL_CAST
+
```

Comment:
I don't think this is quite right yet, but we'll look at this again after merging as I think the multiprecision case could in general be improved here.